pilot/pkg/bootstrap/mesh.go

```diff
@@ -16,174 +16,88 @@ package bootstrap
 
 import (
 	"fmt"
-	"reflect"
-	"time"
 
 	"github.com/davecgh/go-spew/spew"
 	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
 
+	"istio.io/pkg/filewatcher"
+
 	meshconfig "istio.io/api/mesh/v1alpha1"
-	"istio.io/istio/pilot/cmd"
-	"istio.io/istio/pilot/pkg/model"
-	"istio.io/istio/pilot/pkg/networking/util"
-	kubecontroller "istio.io/istio/pilot/pkg/serviceregistry/kube/controller"
-	"istio.io/istio/pkg/config/mesh"
 	"istio.io/pkg/log"
 	"istio.io/pkg/version"
+
+	kubecontroller "istio.io/istio/pilot/pkg/serviceregistry/kube/controller"
+	"istio.io/istio/pkg/config/mesh"
 )
 
 const (
 	// configMapKey should match the expected MeshConfig file name
 	configMapKey = "mesh"
 )
 
-func (s *Server) initMesh(args *PilotArgs) error {
-	if err := s.initMeshConfiguration(args); err != nil {
-		return fmt.Errorf("mesh config: %v", err)
-	}
-	if err := s.initMeshNetworks(args); err != nil {
-		return fmt.Errorf("mesh networks: %v", err)
-	}
-	return nil
-}
+// initMesh creates the mesh in the pilotConfig from the input arguments.
+func newMeshWatcher(args *PilotArgs, kubeClient kubernetes.Interface,
```

Comment:
I'm not sure what is changing here - but the current default is to use the file. 

It is critical to keep it this way since we still want to support working without k8s, and this is the 
most stable/tested code path.

Reply:
(just looking for confirmation this is the intent of the change - hard to follow )

Reply:
The order is:

1. if a `*meshconfig.MeshConfig` was supplied in the arguments, use that.
2. if a file was provided and currently available, use that.
3. use a default mesh config.